### ISSUE TITLE
feat: Figure out how to log *ALL* gRPC requests to the IOx server, even those whose payloads are invalid

### ISSUE THREAD

**Issue by alamb** [influxdata/influxdb_iox#514]
__Use case:__
  There are several bugs such as https://github.com/influxdata/influxdb_iox/issues/346 and https://github.com/influxdata/influxdb_iox/issues/491 where the use of `0xff` is used as a tag key name, and is sent in a field marked as `string` type in protobuf messages. Apparently Golang is cool with this but Rust throws a Kanipshin as it can't make 0xff a valid `String`, seeing as it isn't UTF8. In both these cases, the error decoding the message was not logged in the IOx logs. In #346 the error was displayed by the UI but in #491 I think flux ignored the error.

User story:
  * As someone operating IOx (either in development or production)
  * I want to know exactly what requests were received
  * So that I can troubleshoot what component is having issues

__Proposal:__
Ensure that all requests received by IOx are logged

__Current behaviour:__
Some gRPC requests (I think) correctly return an error to the requestor, but are not logged in the IOx logs

__Desired behaviour:__
 **All** gRPC requests are logged (especially those that error), including when the request can not be deocded correctly (as in #491)

__Alternatives considered:__
I just want my logs!

**Comment by jacobmarble** [comment]
@alamb can this be closed?

**Comment by alamb** [comment]
I don't think we do this yet (though maybe there is some magic in the `istio` sidecar that tracks such requests in our cloud
